bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/bundlefile/CloseableBundleFile.java

```diff
@@ -141,8 +142,21 @@ private void open(boolean keepLock) throws IOException {
 			} else {
 				mruListUse();
 			}
+		} catch (Error | RuntimeException e) {
```

Comment:
This is not enough. You need to catch Throwable since a checked exception can be thrown despite the compiler not thinking so.

Reply:
OK, I force pushed a change that moves the handling of all Throwable out to lockOpen() method.